src/runtime/swg-client-event-manager.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import {isObject, isFunction, isEnumValue} from '../utils/types';
+import * as EventManagerApi from '../api/swg-client-event-manager-api';
+
+/**
+ * Helper function to describe an issue with an event object
+ * @param {!string} valueName
+ * @param {?*} value
+ * @returns {!string}
+ */
+function createEventErrorMessage(valueName, value) {
+  return 'An SwgClientEvent has an invalid ' + valueName + '(' + value + ')';
+}
+
+/**
+ * Throws an error if the event is invalid.
+ * @param {!EventManagerApi.SwgClientEvent} event
+ * @returns {!Promise}
+ */
+function validateEvent(event) {
+  if (!event) {
+    throw new Error('SwgClientEventManager cannot log a null event');
+  }
+
+  if (!isEnumValue(AnalyticsEvent, event.eventType)) {
+    throw new Error(createEventErrorMessage('eventType', event.eventType));
+  }
+  if (!isEnumValue(EventOriginator, event.eventOriginator)) {
+    throw new Error(createEventErrorMessage('eventOriginator',
+        event.eventOriginator));
+  }
+  if (!event.additionalParameters) {
+    event.additionalParameters = {};
+  }
+  if (!isObject(event.additionalParameters)) {
+    throw new Error(createEventErrorMessage('additionalParameters',
+        event.additionalParameters));
+  }
+  if (event.isFromUserAction !== null
```

Comment:
this lines of code is confusing. if you are simply trying to ensure that if this value is not null, its a bool, why not add to utils and use it here?

Reply:
Ok.  I did decide to keep the null check outside of the isBoolean check though.